server/sonar-server/src/main/java/org/sonar/server/platform/db/migration/DatabaseMigrationImpl.java

```diff
@@ -99,22 +98,22 @@ private void startAsynchronousDBMigration() {
   }
 
   private void doDatabaseMigration() {
-    status = Status.RUNNING;
-    startDate = new Date();
-    failureError = null;
+    migrationState.setStatus(Status.RUNNING);
+    migrationState.setStartedAt(new Date());
+    migrationState.setError(null);
     Profiler profiler = Profiler.create(LOGGER);
     try {
       profiler.startInfo("Starting DB Migration");
       doUpgradeDb();
       doRestartContainer();
       doRecreateWebRoutes();
-      status = Status.SUCCEEDED;
+      migrationState.setStatus(Status.SUCCEEDED);
       profiler.stopInfo("DB Migration ended successfully");
     } catch (Throwable t) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1181)